### ISSUE TITLE
[BUG] .gitignore and .npmignore will not ignore a folder if such folder is named nodejs

### ISSUE THREAD

**Issue by onzag** [npm/cli#2139]
### Current Behavior:
If my folder in the ignore repository is named nodejs it is not ignored for pushing

### Expected Behavior:
I expect the folder to be ignored.

### Steps To Reproduce:
You can use my repo to reproduce it, this branch specifically https://github.com/onzag/itemize/tree/006_slate-editor-support

After you npm install and npm run install, try the dry run for packing and you realize that the nodejs folder is included regardless of what .gitignore says

Tested in both npm 6 and npm 7

**Comment by ljharb** [comment]
I tried this as well, and was able to reproduce it - naming the folder "node" or "tmp" both caused it to be properly ignored. It seems perhaps that the "nodejs" folder name is magic?

**Comment by isaacs** [comment]
There's nothing magical about the name `nodejs`.

When I run `npm pack` on the branch specified with either npm 7 or npm 6, I don't get a `nodejs` folder in the tarball.

You just have your bin set to a file in the `nodejs` folder on your master branch, maybe that's it?

**Comment by ljharb** [comment]
@isaacs i was able to reproduce this locally by cloning this repo - https://github.com/onzag/itemize/blob/006_slate-editor-support/package.json#L6 isn't pointing to anything inside the `nodejs` folder (altho https://github.com/onzag/itemize/blob/006_slate-editor-support/package.json#L9 is, but npm presumably ignores the `types` field).

Since i was able to repro it locally, is there something I can run that would help debug it?

**Comment by isaacs** [comment]
Here's what I see:

```
$ git clone git@github.com:onzag/itemize.git -b 006_slate-editor-support
Cloning into 'itemize'...
remote: Enumerating objects: 503, done.
remote: Counting objects: 100% (503/503), done.
remote: Compressing objects: 100% (318/318), done.
remote: Total 19367 (delta 273), reused 346 (delta 179), pack-reused 18864
Receiving objects: 100% (19367/19367), 5.81 MiB | 5.75 MiB/s, done.
Resolving deltas: 100% (14387/14387), done.
Updating files: 100% (426/426), done.

$ cd itemize/

$ npm pack
npm notice
npm notice 📦  @onzag/itemize@0.0.114.slate
npm notice === Tarball Contents ===
...
npm notice === Tarball Details ===
npm notice name:          @onzag/itemize
npm notice version:       0.0.114.slate
npm notice filename:      @onzag/itemize-0.0.114.slate.tgz
npm notice package size:  720.7 kB
npm notice unpacked size: 3.1 MB
npm notice shasum:        267cf4400c2800632c93551e6ca5ff104299a3e8
npm notice integrity:     sha512-3TsP5KpCL7Gnk[...]hz3hVfLfDDYGw==
npm notice total files:   424
npm notice
onzag-itemize-0.0.114.slate.tgz

$ tar tvf onzag-itemize-0.0.114.slate.tgz | grep nodejs

```

Are you sure that you're not looking at a tarball created from the master branch?

**Comment by ljharb** [comment]
Ah, you're right - i was indeed checking on master.

**Comment by onzag** [comment]
@isaacs 

https://github.com/onzag/itemize/blob/master/.gitignore

Master branch contains exactly the same gitignore and works the same way as that branch, they are not very different; that branch that I sent to ljharb was just because it was what I was working in the moment but it should make no difference.

Also `npm install` had to be ran before packing.

That said I started getting the same results you do as well and suddenly it stopped appearing in the tarball; I don't understand, but @ljharb replicated it as well, that was very strange.

What I did was that on the npm registry I pushed a version that had no "nodejs" folder in it, I manually deleted the folder and published a new version, and after that it stopped showing the nodejs and started being properly ignored.

All my older versions contained the folder nodejs in it since they weren't being ignored.
